nncf/torch/graph/patterns.py

```diff
@@ -11,27 +11,131 @@
+    bn_node = pattern.add_node(BATCH_NORMALIZATION_OPERATIONS)
+    sigmoid_node = pattern.add_node(['sigmoid'])
+    mul_node = pattern.add_node(['__mul__'])
 
-ACTIVATIONS = RELU | NON_RELU_ACTIVATIONS
+    pattern.add_edge(linear_ops_node, bn_node)
+    pattern.add_edge(bn_node, sigmoid_node)
+    pattern.add_edge(linear_ops_node, mul_node)
 
-ANY_BN_ACT_COMBO = BN + ACTIVATIONS | ACTIVATIONS + BN | BN | ACTIVATIONS
+    main_pattern.add_pattern_alternative(pattern)
 
-SINGLE_OPS = ACTIVATIONS | POOLING | N('mean') | N('layer_norm')
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    sigmoid_node = pattern.add_node(['sigmoid'])
+    mul_node = pattern.add_node(['__mul__'])
 
-ARITHMETIC = N('__iadd__') | N('__add__') | N('__mul__') | N('__rmul__')
+    pattern.add_edge(linear_ops_node, sigmoid_node)
+    pattern.add_edge(linear_ops_node, mul_node)
 
-ELTWISE_UNIFORM_OPS = BN | RELU | ACTIVATIONS
+    main_pattern.add_pattern_alternative(pattern)
 
-MATMUL = N('matmul') | N('bmm')
+    return main_pattern
+
+
+def create_linear_h_swish_act() -> GraphPattern:
+    main_pattern = GraphPattern()
+
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    bn_node = pattern.add_node(BATCH_NORMALIZATION_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
+    mul_node = pattern.add_node(['__mul__'])
+
+    pattern.add_edge(linear_ops_node, bn_node)
+    pattern.add_edge(bn_node, __add__node)
+    pattern.add_edge(bn_node, mul_node)
+    pattern.add_edge(__add__node, hardtanh_node)
+    pattern.add_edge(hardtanh_node, __truediv__node)
+    pattern.add_edge(__truediv__node, mul_node)
+
+    main_pattern.add_pattern_alternative(pattern)
+
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
+    mul_node = pattern.add_node(['__mul__'])
+
+    pattern.add_edge(linear_ops_node, __add__node)
+    pattern.add_edge(linear_ops_node, mul_node)
+    pattern.add_edge(__add__node, hardtanh_node)
+    pattern.add_edge(hardtanh_node, __truediv__node)
+    pattern.add_edge(__truediv__node, mul_node)
+
+    main_pattern.add_pattern_alternative(pattern)
+
+    return main_pattern
+
+
+def create_linear_h_sigmoid_act() -> GraphPattern:
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
+
+    pattern.add_edge(linear_ops_node, __add__node)
+    pattern.add_edge(__add__node, hardtanh_node)
+    pattern.add_edge(hardtanh_node, __truediv__node)
+
+    return pattern
+
```

Comment:
Should separate the swish and h_swish into their own patterns, instead of always pairing them with linear or BN. We will need to fuse swish and h_swish even if they appear in the graph on their own.

Reply:
I also was thinking about that. But in swish and h_swish we have a starting node from we calculate sigmoid and then __mul__ the result from sigmoid and the initial node. It doesn't seem solvable for me now

Reply:
Then you have to implement special "PatternInput" and "PatternOutput" nodes, and remove these from the composite graph pattern during the "+" or "join_patterns" operation.

Reply:
Thanks for the idea! I added a node with a special type = 'INPUT_NODE'. This node implements the logic that you described. "+" operation changes this node on the last one from the other graph. I will add the same approach to "join_patterns"

Reply:
`INPUT_NODE` is not a good name since we also have input nodes and output nodes in the NNCFGraph, and this name could be mistaken as to be referring to the model inputs and outputs. Use `PATTERN_INPUT`.

Reply:
Updated

Reply:
> `INPUT_NODE` is not a good name since we also have input nodes and output nodes in the NNCFGraph, and this name could be mistaken as to be referring to the model inputs and outputs. Use `PATTERN_INPUT`.

@vshampor, if you introduce `GraphPattern.PATTERN_INPUT_NODE_TYPE` then `PATTERN_` is definitely a noise here which just makes the names loner.